CPAC/pipeline/engine.py

```diff
@@ -1371,6 +1376,20 @@ def gather_pipes(self, wf, cfg, all=False, add_incl=None, add_excl=None):
                 write_json.inputs.json_data = json_info
 
                 wf.connect(id_string, "out_filename", write_json, "filename")
+
+                # Node to validate TR (and other scan parameters)
+                validate_bold_header = pe.Node(
+                    Function(
+                        input_names=["input_bold", "RawSource_bold"],
+                        output_names=["output_bold"],
+                        function=validate_outputs,
+                        imports=[
+                            "from CPAC.pipeline.utils import find_pixdim4, update_pixdim4"
+                        ],
+                    ),
+                    name=f"validate_bold_header_{resource_idx}_{pipe_x}",
+                )
```

Comment:
I think this could go inside this https://github.com/FCP-INDI/C-PAC/blob/d078863085de062b66c996d5f8beca95ea4ab9d9/CPAC/pipeline/engine.py#L1416 block since we're only using this Node on `*_bold` resources, we don't need to create one for any other resources

Reply:
Yup good idea.